app/models/ipn.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class Ipn < ApplicationRecord
+  enum status: {
+    pending:   'pending',
+    processed: 'processed',
+    error:     'error'
+  }
+
+  scope :mercado_pago, -> { where gateway: 'mercado_pago' }
+
+  def process!
+    processed = case gateway
+                when 'mercado_pago'
+                  MercadoPagoClient.process_ipn self
+                end
+
+    update status: (processed ? :processed : :error)
```

Comment:
Si el método es `process!` acá deberíamos llamar a:
```suggestion
    update! status: (processed ? :processed : :error)
```

Aunque si no hay un `process`, preferiría sin el `!`. Igual dejaría el `update!` porque como esto va a estar en una tarea es preferible que explote a que devuelva un `false` que ni revisamos.

Reply:
Dale x2